# Setting up

## Access to Xanadu

The workshop will use resources available at UConn's HPC Xanadu.

How will people log in? This is something you can sort out with Noah

```
ssh username@xanadu-submit-ext.cam.uchc.edu
```

## Genomic resources

We will be using chromosome 1 from the Boxelder (Acer negundo) for this workshop ([McEvoy et al. 2022](https://onlinelibrary.wiley.com/doi/10.1111/tpj.15657)). The assembly was softmasked using RepeatModeler v2.01 ([Flynn et al. 2020](https://doi.org/10.1073/pnas.1921046117)) and RepeatMasker v4.0.6 ([Smith et al. 2013](http://www.repeatmasker.org)).

The sequences are avaialble in the workshop github (data/genome/chr1.fna) or can be downloaded from GenBank:

```
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/025/594/385/GCA_025594385.1_ASM2559438v1/GCA_025594385.1_ASM2559438v1_genomic.fna.gz
```

To unzip the file, use gunzip

```
gunzip GCA_025594385.1_ASM2559438v1_genomic.fna.gz
```

The fasta file is interleaved and sequences are organized by scaffold. To isoloate chromosome 1 we can determine the number of lines and use `head` to subset the file. 

First, get the number of sequences in each chromosome. There are 13 chromosomes in the Boxelder genome, and we can ignore the other scaffolds for now.

```
awk '/^>/ {if (seqlen){print seqlen}; print ;seqlen=0;next; } { seqlen = seqlen +length($0)}END{print seqlen}' GCA_025594385.1_ASM2559438v1_genomic.fna | head -n26
```
The output shows chromosome 1 contains 51671011 sequences. 

```
>CM046694.1 Acer negundo isolate 91603 chromosome 1, whole genome shotgun sequence
51671011
>CM046695.1 Acer negundo isolate 91603 chromosome 2, whole genome shotgun sequence
49136802
>CM046696.1 Acer negundo isolate 91603 chromosome 3, whole genome shotgun sequence
41658500
>CM046697.1 Acer negundo isolate 91603 chromosome 4, whole genome shotgun sequence
37510000
>CM046698.1 Acer negundo isolate 91603 chromosome 5, whole genome shotgun sequence
33812531
>CM046699.1 Acer negundo isolate 91603 chromosome 6, whole genome shotgun sequence
32304000
>CM046700.1 Acer negundo isolate 91603 chromosome 7, whole genome shotgun sequence
30451126
>CM046701.1 Acer negundo isolate 91603 chromosome 8, whole genome shotgun sequence
30120719
>CM046702.1 Acer negundo isolate 91603 chromosome 9, whole genome shotgun sequence
29334433
>CM046703.1 Acer negundo isolate 91603 chromosome 10, whole genome shotgun sequence
27516629
>CM046704.1 Acer negundo isolate 91603 chromosome 11, whole genome shotgun sequence
27372411
>CM046705.1 Acer negundo isolate 91603 chromosome 12, whole genome shotgun sequence
26422946
>CM046706.1 Acer negundo isolate 91603 chromosome 13, whole genome shotgun sequence
23969194
```

The fasta file is folded at 80 sequences and we know chromosome 1 contains 51671011 sequences. Using a simple calculation of the number of sequences divided by the number of sequences on a line (51671011/80 = 645887.6) we can determine how many lines are needed. Include an extra line for the header. 

```
head -n645889 GCA_025594385.1_ASM2559438v1_genomic.fna > chr1.fna
```

We can check that `chr1.fna` contains the expected number of sequences

```
awk '/^>/ {if (seqlen){print seqlen}; print ;seqlen=0;next; } { seqlen = seqlen +length($0)}END{print seqlen}' chr1.fna
>CM046694.1 Acer negundo isolate 91603 chromosome 1, whole genome shotgun sequence
51671011
```

and it does. An alternative method to extract chromosome 1 is with [seqkit's grep](https://bioinf.shenwei.me/seqkit/usage/#grep). Seqkit is installed on Xanadu and can be loaded and used with

```
module load seqkit/2.2.0
seqkit grep -i -r -p 'CM046694.1+$' GCA_025594385.1_ASM2559438v1_genomic.fna  -o out.fna
```

### Trascriptomic data

RNA sequences will be used to idenity gene models. To obtain the sequneces, EASEL will download them with the sratoolkit from a list of run id's (e.g. SRR-id's). The SRA list for Boxelder is located at `data/sra/acer_negundo.txt`. 

## Software

Brief overview of the software that we will use. (very brief, more of a list with citations.)


## References

1. [McEvoy, Susan L., et al. "Strategies of tolerance reflected in two North American maple genomes." The Plant Journal 109.6 (2022): 1591-1613.](https://onlinelibrary.wiley.com/doi/abs/10.1111/tpj.15657)
2. [Flynn, Jullien M., et al. "RepeatModeler2 for automated genomic discovery of transposable element families." Proceedings of the National Academy of Sciences 117.17 (2020): 9451-9457.](https://doi.org/10.1073/pnas.1921046117)
3. [Smit, A., Hubley, R. & Green, P. (2013) RepeatMasker Open-4.0.](http://www.repeatmasker.org)

