# Introduction to Genome Annotation

*This section is intended to be a brief, but well written and cited section. About four paragraphs with a figure or two that describe genome annotation to the neophyte.*

Annotation of protein-coding genes in an assembled genome is an essential process for genome sequencing projects. Structurally annotatted eukaryotic gene models provide precise information regarding the location of a gene on a chromosome, translation initiation sites, untranslated (UTR) regions, and critically, introns and exons containing coding sequences, among other features ([Yandel & Ence, 2012](https://www.nature.com/articles/nrg3174)). An annotated genome enables numerous inquiries into genetic and genomic questions, and thus are a critical building block in the biological sciences. 

Annotating gene models in non-model organisms is a challenging task ([Yandel & Ence, 2012](https://www.nature.com/articles/nrg3174)). In contrast to model organisms, non-models typically lack robust knowledge of gene models gained from decades of sanger sequencing, genetic modification of entire genes (e.g. knock-out studies), or genetic modification of single exons or sites. The information gained from classical genetic studies allows for robust identification of gene model structures (e.g. exon/intron boundaries). In non-model systems, methods for gene model prediction include *ab initio*, comparative, homology-based, or a combination of these methods ([Gremme, 2012](https://ediss.sub.uni-hamburg.de/handle/ediss/4964)).

*Ab initio* gene model predictions is among the most popular method for genome annotation in non-model organisms. In *ab initio* annotation, gene models are predicted using genomic sequences and a model fit containing information on gene structures in the target genome or a close relative ([Yandel & Ence, 2012](https://www.nature.com/articles/nrg3174)). Gene models are fit using a genome assembly and external evidence describing gene models. The evidence can come in the from of transcriptomes from RNAseq experiments or protein-based evidence containing amino acid sequences. Gene models are created through machine-learning algorithms, such as hidden Markov models (HMMs), by identifying sequence similarity between the genome assembly and external inforamtion ([Scalzitti et al. 2020](https://doi.org/10.1186/s12864-020-6707-9)). This approach is implemented in software like AUGUSTUS ([Stanke et al. 2006](https://doi.org/10.1186/1471-2105-7-62)) or FGENESH


Merging information from multiple analytical pathways can improve annotation quality, but is conceputally challenging in situations where the truth is not known or not knowable without signficant investment in human resources. ([Yandel & Ence, 2012](https://www.nature.com/articles/nrg3174)). 




### References

1. [Yandell, Mark, and Daniel Ence. "A beginner's guide to eukaryotic genome annotation." Nature Reviews Genetics 13.5 (2012): 329-342.](https://www.nature.com/articles/nrg3174).
2. [Gremme, Gordon. Computational gene structure prediction. Diss. Staats-und Universit√§tsbibliothek Hamburg Carl von Ossietzky, 2012.](https://ediss.sub.uni-hamburg.de/handle/ediss/4964)