# Introduction to Genome Annotation

*This section is intended to be a brief, but well written and cited section. About four paragraphs with a figure or two that describe genome annotation to the neophyte.*

Annotation of protein-coding genes in an assembled genome is an essential process for genome sequencing projects. Structurally annotatted eukaryotic gene models provide precise information regarding the location of a gene on a chromosome, translation initiation sites, untranslated (UTR) regions, and critically, introns and exons containing coding sequences, among other features ([Yandel & Ence, 2012](https://www.nature.com/articles/nrg3174)). An annotated genome enables numerous inquiries into genetic and genomic questions, and thus is a critical building block for biological sciences. 

Annotating gene models in non-model organisms is a challenging task, particularly in non-model eukarytotes which have more complicated gene structures than prokaryotes ([Yandel & Ence, 2012](https://www.nature.com/articles/nrg3174)) Non-models eukaryotes typically lack robust knowledge of gene models gained from decades of sanger sequencing, genetic modification of entire genes (e.g. knock-out studies), or genetic modification of single exons or sites that are routinely performed in model organisms. 

Genome annotaiton is a process that involves multiple software and decisions by the user that influence the quality of the output. The difficulty of annotation most often lies in the fact that the end result is unknown and potentially unkowable. Heuristic methods relying on positive or negative comparison to a reference can't be performed unless a reference annotation is already available. Although, there are many ad hoc evalution statistics that can assess annotaiton quality, including BUSCO [Simao et al. 2015](https://academic.oup.com/bioinformatics/article/31/19/3210/211866), the ratio of mono- to multi-exonic genes, or EnTAP gene prediction qulaity ([Vuruputoor et al. 2022](https://doi.org/10.1101/2022.10.03.510643)). These evulation methods can guide the selection of tools and outcomes in an annotation pipeline. 

Genome annotations fall into three broad categories: repeat annotation, structural annotaiton, and functional annotation. Before genome annotaiton can begin, an asssembled genome is required (tutorials for genome assembly can be found [here](https://bioinformatics.uconn.edu/resources-and-events/tutorials-2/)). Repeat annotaiton identifies repetitive elements in the genome which are then masked during down stream analyses. For projects interested in protein-coding genes, repetitive elements introduce uncertainity and masking them imrpoves annotation quality. RepeatModeler2 ([Flynn et al. 2020](https://www.pnas.org/doi/10.1073/pnas.1921046117)) or RepeatMasker ([Smit et al. 2013](https://www.repeatmasker.org/)) are common tools for repeat masking. 

In non-model systems, methods for gene  prediction include *ab initio*, comparative, homology-based, or a combination of these methods ([Gremme, 2012](https://ediss.sub.uni-hamburg.de/handle/ediss/4964)). *Ab initio* gene model prediction is among the most popular method for genome annotation in non-model organisms. In *ab initio* annotation, gene models are predicted using genomic sequences and a model fit containing information on gene structures in the target genome or a close relative ([Yandel & Ence, 2012](https://www.nature.com/articles/nrg3174)). Gene models are fit using a genome assembly and external evidence describing gene models. The evidence can come in the from of transcriptomes from RNAseq experiments or protein-based evidence containing amino acid sequences. Gene models are created through machine-learning algorithms, such as hidden Markov models (HMMs), by identifying sequence similarity between the genome assembly and external inforamtion ([Scalzitti et al. 2020](https://doi.org/10.1186/s12864-020-6707-9)). This approach is implemented in software like AUGUSTUS ([Stanke et al. 2006](https://doi.org/10.1186/1471-2105-7-62)) or FGENESH ([Salamov and Solovyev 2000](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC310882/)).


Merging information from multiple analytical pathways can improve annotation quality, but is conceputally challenging in situations where the truth is not known or not knowable without signficant investment in human resources. ([Yandel & Ence, 2012](https://www.nature.com/articles/nrg3174)). 

Combining protein and RNAseq evidence makes the annotation worse (Vidya's paper). However, filtering annotations based on support metrices can improve annotation quality when assessed by F1 score (refer to benchmarking.)

Structural annotation vs. functional annotation. Describe functional annotation next.

*Get basic here, have some figures and cartoons that really expalin the process*


### Notes from Readings

Features that are important for corretly identifying gene models are gene length, presence of start codon, melting temperature, and GC content ([Upadhyaya et al. 2022](https://pubmed.ncbi.nlm.nih.gov/35736770/)).

AUGUSTUS is the best ab initio gene model annotator based on overall performance of correctly annotating genes at the nucleotide and protein level ([Scalzitti et al. 2020](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-020-6707-9)). 

However, the relative performance of ab initio annotators is complex and evaluation of benchmarking results presents a nuanced picture. A benchmarking case study using 1793 protein coding genes demonstrates that while AUGUSTUS performs best overall, all popular methods have difficulty identifying protein-coding genes  ([Scalzitti et al. 2020](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-020-6707-9)). 



### References

1. [Yandell, Mark, and Daniel Ence. "A beginner's guide to eukaryotic genome annotation." Nature Reviews Genetics 13.5 (2012): 329-342.](https://www.nature.com/articles/nrg3174).
2. [Gremme, Gordon. Computational gene structure prediction. Diss. Staats-und Universitätsbibliothek Hamburg Carl von Ossietzky, 2012.](https://ediss.sub.uni-hamburg.de/handle/ediss/4964)
3. [Simão, Felipe A., et al. "BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs." Bioinformatics 31.19 (2015): 3210-3212.](https://academic.oup.com/bioinformatics/article/31/19/3210/211866)
4. []()