# Introduction to Genome Annotation

Annotation of protein-coding genes in an assembled genome is an essential process for genome sequencing projects. Structurally annotatted eukaryotic gene models provide precise information regarding the location of a gene on a chromosome, translation initiation sites, untranslated (UTR) regions, and critically, introns and exons containing coding sequences, among other features ([Yandel & Ence, 2012](https://www.nature.com/articles/nrg3174)). Knowledge of gene models enables numerous methods and inquiries into genetic and genomic questions, and is thus a critical building block for the biological sciences. 

Annotating gene models is a challenging task, particularly in non-model eukarytotes which have more complicated gene structures than prokaryotes ([Yandel & Ence, 2012](https://www.nature.com/articles/nrg3174)). Non-model eukaryotic systems typically lack robust knowledge of gene model structure gleaned from decades of genetic research. Studies in model systems using methods such as sanger sequencing, genetic modification of entire genes (e.g. knock-out studies), modification of specific exons or sites, and numerous other genetic methods, allow for precise manual curation of gene model annotaitons.

In non-model organisms where it is impracical to employ sophisticated genetic methods, genome annotaiton is carried out with multiple software using a combination of genomic and transcriptomic data. The difficulty of annotation most often lies in the fact that the end result is unknown and potentially unkowable. Heuristic methods relying on positive or negative comparison to a reference can't be performed unless a reference annotation is already available. Although, there are many ad hoc evalution statistics that can assess annotaiton quality, including BUSCO ([Simao et al. 2015](https://academic.oup.com/bioinformatics/article/31/19/3210/211866)), the ratio of mono- to multi-exonic genes, or EnTAP gene prediction quality ([Vuruputoor et al. 2022](https://doi.org/10.1101/2022.10.03.510643)). These evaluation methods can guide the selection of tools and outcomes during an annotation pipeline. 

Genome annotations fall into three broad categories: repeat annotation, structural annotaiton, and functional annotation. Before genome annotaiton can begin, an asssembled genome is required (tutorials for genome assembly can be found [here](https://bioinformatics.uconn.edu/resources-and-events/tutorials-2/)). Repeat annotaiton identifies repetitive elements in the genome which are then masked during down stream analyses. For projects interested in protein-coding genes, repetitive elements introduce uncertainity and masking them imrpoves annotation quality. RepeatModeler2 ([Flynn et al. 2020](https://www.pnas.org/doi/10.1073/pnas.1921046117)) or RepeatMasker ([Smit et al. 2013](https://www.repeatmasker.org/)) are common tools for repeat masking. 

In non-model systems, methods for gene  prediction include *ab initio*, comparative, homology-based, or a combination of these methods ([Gremme, 2012](https://ediss.sub.uni-hamburg.de/handle/ediss/4964)). *Ab initio* gene model prediction is among the most popular methods for genome annotation in non-model organisms. In *ab initio* annotation, gene models are predicted using genomic sequences and a model fit containing information on gene structures in the target genome or a close relative ([Yandel & Ence, 2012](https://www.nature.com/articles/nrg3174)). Gene models are fit using a genome assembly and external evidence describing gene models. The evidence can come in the from of transcriptomes from RNAseq experiments or protein-based evidence containing amino acid sequences. Gene models are created through machine-learning algorithms, such as hidden Markov models (HMMs), by identifying sequence similarity between the genome assembly and external inforamtion ([Scalzitti et al. 2020](https://doi.org/10.1186/s12864-020-6707-9)). This approach is implemented in software like AUGUSTUS ([Stanke et al. 2006](https://doi.org/10.1186/1471-2105-7-62)) or FGENESH ([Salamov and Solovyev 2000](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC310882/)). Currently, AUGUSTUS is widely considered the best *ab initio* annotator based on overall performance of correctly annotating genes at the nucleotide and protein level ([Scalzitti et al. 2020](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-020-6707-9)). However, the relative performance of *ab initio* annotators is complex and evaluation of benchmarking results presents a nuanced picture. A benchmarking case study using 1793 protein coding genes demonstrates that while AUGUSTUS performs best overall, all popular methods have difficulty identifying protein-coding genes ([Scalzitti et al. 2020](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-020-6707-9)). 

Merging information from multiple analytical pathways to improve annotation quality has been proposed and successfully implemented in some pipelines (e.g. TSEBRA ([Gabriel et al. 2021](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04482-0))). Approaches which combine external evidence, like protein and RNAseq or long and short-read RNAseq information, may improve annotation quality; however, meta-analyses indicate there are numerous opportunities for combinatorial approaches to fail and generate worse annotations than more simple approaches ([Vuruputoor et al. 2022](https://doi.org/10.1101/2022.10.03.510643)). Another approach to combine information relies on identification of gene model features from multiple annotations and creates a matrix of feature weights for each putative gene model. The weights matrix is filtered based on criteria, yielding a putative set of the best supported gene models. Gene length, presence of a start codon, melting temperature, and GC content have been identified as the most reliable gene features ([Upadhyaya et al. 2022](https://pubmed.ncbi.nlm.nih.gov/35736770/)), but this is an active area of research.

The third broad category of genome annotations associates biological information to protein-coding genes or regulatory RNAs to provide functional information. Gene function predicitons are made through sequence similarity to databases of genes with known functions ([Ejigu and Jung, 2020](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7565776/)). A critical assumption of these methods is that gene function is conserved evolutionarily and orthologs and paralogs will more than likely maintain their function across species. Functional annotaitons include gene ontology (GO) terms describing molecular function, the cellular component and biological process where the protein is active. Numerous other databases provide functional information and are reviewed elsehwere ([Ejigu and Jung, 2020](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7565776/)). Software for functional annotation includes [EnTAP](https://entap.readthedocs.io/en/v0.8.0-beta/introduction.html) ([Hart et al. 2020](https://doi.org/10.1111/1755-0998.13106)) and BLAST2GO ([Conesa et al., 2005](https://doi.org/10.1093/bioinformatics/bti610)), among others.

In this workshop, we will be using the EASEL genome annotaiton pipeline to generate structural genome annotaitons. EASEL relies on RNAseq data for deriving gene models as extrinsic evidence to AUGUSTUS. Predicted gene structures are compared with the gene prediction generated using popular structural genome annotation tools, such as, BRAKER and FINDER. The predicted gene structures from different methods are evaluated with estimating F1 score (harmonic mean) for the genic features including exon, transcript, and gene.

### References

1. [Yandell, Mark, and Daniel Ence. "A beginner's guide to eukaryotic genome annotation." Nature Reviews Genetics 13.5 (2012): 329-342.](https://www.nature.com/articles/nrg3174).
2. [Simão, Felipe A., et al. "BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs." Bioinformatics 31.19 (2015): 3210-3212.](https://academic.oup.com/bioinformatics/article/31/19/3210/211866)
3. [Vuruputoor, Vidya Sudershan, et al. "Welcome to the big leaves: best practices for improving genome annotation in non-model plant genomes." bioRxiv (2022).](https://www.biorxiv.org/content/10.1101/2022.10.03.510643v2)
4. [Gremme, Gordon. Computational gene structure prediction. Diss. Staats-und Universitätsbibliothek Hamburg Carl von Ossietzky, 2012.](https://ediss.sub.uni-hamburg.de/handle/ediss/4964)
4. [Gabriel, Lars, et al. "TSEBRA: transcript selector for BRAKER." BMC bioinformatics 22.1 (2021): 1-12.](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04482-0)
6. [Ejigu, Girum Fitihamlak, and Jaehee Jung. "Review on the computational genome annotation of sequences obtained by next-generation sequencing." Biology 9.9 (2020): 295.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7565776/)
8. [Hart, Alexander J., et al. "EnTAP: Bringing faster and smarter functional annotation to non‐model eukaryotic transcriptomes." Molecular ecology resources 20.2 (2020): 591-604.](https://doi.org/10.1111/1755-0998.13106)
9. [Conesa, Ana, et al. "Blast2GO: a universal tool for annotation, visualization and analysis in functional genomics research." Bioinformatics 21.18 (2005): 3674-3676.](https://doi.org/10.1093/bioinformatics/bti610)