# Running EASEL

EASEL is compiled as a nextflow script and is run by submitting a shell script to a job handler (e.g. slrum). Two files are required in a directory to run EASEL, a job submission script (`nextflow.sh`) and the starting nextflow script (`main.nf`) Edit path information, supplying the requreied inputs (genome, RNAseq reads, and outdir), and submit the job. 


An EASEL run using the default parameters would have this statement in a bash script with the appropraite header for your job handler

```
Usage:
nextflow main.nf --genome /path/to/genome.fa --outdir /path/to/outdir --sra /path/to/sralist.txt [...] 

```

Submitting EASEL to the slurm job handler on Xanadu would use a script like the following, called `nextflow.sh`

```
#!/bin/bash
#SBATCH --job-name=species
#SBATCH -N 1
#SBATCH -n 1
#SBATCH -c 4
#SBATCH --partition=general
#SBATCH --qos=general
#SBATCH --mem=10G
#SBATCH --mail-user=user@uconn.edu

###Load nextflow
module load nextflow

###Activate your AGAT environment###
source activate AGAT

nextflow run -w work_cynthia main.nf --species species_nf --outdir species_easel --sra /path/to/sra/species.txt --genome /path/to/genome.masked.fa

```

Submiting the job is as simple as 

```
sbatch nextflow.sh
```

You can monitor the progress of the nextflow run with `squeue`, or view the more detailed progress of the job in the `slurm-JOBID.out` file.

The nextflow.sh file edits the config file. After running the job it will create a directory structure with these folders and files.

```
show a tree of the files
```



Show the two places to set the config file, the config file itslef, and how to launch nextflow on Xanadu

## Nextflow Config file

## Second config file

