# Running EASEL

EASEL is compiled as a nextflow script and is run by submitting a shell script to a job handler (e.g. slrum). The files and environments for running EASEL have been copied to your folder on Xanadu located at `/home/FCAM/easel/`. Two files are required in a directory to run EASEL, a job submission script (`nextflow.sh`) and the starting nextflow script (`main.nf`) Edit path information, supplying the requreied inputs (genome, RNAseq reads, and outdir), and submit the job. 

An EASEL run using the default parameters would have this statement in a bash script with the appropraite header for your job handler

```
Usage:
nextflow main.nf --genome /path/to/genome.fa --outdir /path/to/outdir --sra /path/to/sralist.txt [...] 

```


### nextflow.sh

Submitting EASEL to the slurm job handler on Xanadu would use a script like the following, called [`nextflow.sh`](files/nextflow.sh)

```
#!/bin/bash
#SBATCH --job-name=species
#SBATCH -N 1
#SBATCH -n 1
#SBATCH -c 4
#SBATCH --partition=general
#SBATCH --qos=general
#SBATCH --mem=10G
#SBATCH --mail-user=user@uconn.edu

###Load nextflow
module load nextflow

###Activate your AGAT environment###
source activate AGAT

nextflow run -w work_cynthia main.nf --species species_nf --outdir species_easel --sra /path/to/sra/species.txt --genome /path/to/genome.masked.fa

```

Submiting the job is as simple as 

```
sbatch nextflow.sh
```

You can monitor the progress of the nextflow run with `squeue`, or view the more detailed progress of the job in the `slurm-JOBID.out` file.

The nextflow.sh file edits the config file. After running the job it will create a directory structure with these folders and files.

```
show a tree of the files
```

Show the two places to set the config file, the config file itslef, and how to launch nextflow on Xanadu

### nextflow.config

The [`nextflow.config`](files/nextflow.config) file contains information on the job handler of the HPC, an ordered list of processes in the pipeline and paths to their modules on Xanadu, and a set of parameters regulating the behavior of EASEL. A user may want to change the items in the parameter block based on their input data (e.g. to loosen or tighten QC metrics).

```
params {
    outdir                   = './easel_results'
    genome                   = '/core/labs/Wegrzyn/easel/data/genomes/Yeast_S288C_reference_sequence_R64-3-1_20210421.fa'
    scripts                  = '/core/labs/Wegrzyn/easel/nextflow/scripts'
    reference                = '/core/labs/Wegrzyn/easel/data/gtf/Physcomitrium_patens.Phypa_V3.52.chr.gtf'
    bins                     = '50'
    sra                      = 'false' 
    user_reads               = 'false'
    species                  = 'easel' 
    fastp_cpu                = '4'
    rate                     = '85'
    total_reads              = '15000000'
    mean_length              = '70'
    psiclass_cpu             = '8'
    stringtie_cpu            = '8'
    cluster_id               = '0.80'
    configAugustus           = '/core/labs/Wegrzyn/easel/software/Augustus/config'
    bin                      = '/core/labs/Wegrzyn/easel/software/Augustus/bin'
    script                   = '/core/labs/Wegrzyn/easel/software/Augustus/scripts'
    configEST                = '/core/labs/Wegrzyn/EASEL/AUGUSTUS/nextflow/DSL2/TEST/config/est.cfg'
    configProtein            = '/core/labs/Wegrzyn/EASEL/AUGUSTUS/nextflow/DSL2/TEST/config/protein.cfg'
    miniprot                 = '/core/labs/Wegrzyn/easel/software/miniprot'
    N                        = '30'
    outn                     = '1000'
    max_intron               = '200000'
    miniprot_cpu             = '12'
    aligner                  = 'miniprot'
    test                     = '250'
    train                    = '1000'
    orthoDB                  = 'embryophyta_odb10'
    tcoverage                = '50'
    qcoverage                = '50'
    taxon                    = ''
    entap_cpu                = '12'
    gFACs                    = '/core/labs/Wegrzyn/easel/software/gFACs'
    window                   = '400_200'
    parts                    = '5'
}
```

### main.nf

The [`main.nf`](files/main.nf) file contains the arguments to EASEL that are passed on to the nextflow executors. The file is useful as a reference, but not much should be edited here.

```
def helpMessage() {
        log.info"""
        ========================================================================================
                 EASEL - Efficient, Accurate Scalable Eukaryotic modeLs
        ========================================================================================
 
        Usage:
        nextflow main.nf --genome /path/to/gen \ \ome.fa --outdir /path/to/outdir --sra /path/to/sralist.txt [...] 

        Required arguments:
                --genome                                Path to masked genome (*.fa/*.fasta)
                --sra                                   Path to SRA accession list
                  or
                --user_reads                            Path to user RNA reads (*_{1,2}.fastq)

        Optional arguments:
                --outdir                                Path to the output directory (default: easel_results)
                --species                               Species prefix (default: SPECIES)
                --chunks                                Number of chunks genome will be split into (default: 50)
                --rate                                  Filtering mapping rate cut-off (default: 85)
                --length                                Filtering read length cut-off (default: 15000000)
                --cluster_id                            USEARCH identify threshold (default: 0.80)
                --aligner                               Protein alignment tool [gth/miniprot] (default: gth)
                --N                                     Miniprot: retain at most NUM number of secondary chains/alignments (default: 30)
                --outn                                  Miniprot: output up to min{NUM, -N} alignments per query (default: 1000)
                --max_intron                            Miniprot: max intron size (default: 200)
                --fastp_cpu                             Number of threads dedicated to fastp (default: 4)
                --stringtie_cpu                         Number of threads dedicated to stringtie2 (default: 8)
                --psiclass_cpu                          Number of threads dedicated to psiclass (default: 8)
                --miniprot_cpu                          Number of threads dedicated to miniprot (default: 12)
                --test                                                  Number of genes to test AUGUSTUS
                --train                                                 Number of genes to train AUGUSTUS
                -resume                                 Resume script from last step 

        Documentation:
        https://gitlab.com/PlantGenomicsLab/easel-augustus-training

        Authors:
        Cynthia Webster <cynthia.webster@uconn.edu>
        ========================================================================================

    """.stripIndent()
}

params.help = false
if (params.help){
    helpMessage()
    exit 0
}

nextflow.enable.dsl=2

include { main_workflow as MAIN } from './workflows/main_workflow.nf'

workflow {

    MAIN()

    }
```
