## Manual

EASEL is run from scripts that set the paths and behavior of the nextflow runs. The nextflow command to run easel takes the following commands.

```

========================================================================================
                EASEL - Efficient, Accurate Scalable Eukaryotic modeLs
========================================================================================
	
Usage:
nextflow main.nf --genome /path/to/genome.fa --outdir /path/to/outdir --sra /path/to/sralist.txt [...] 

Required arguments:
	--genome				Path to masked genome (*.fa/*.fasta)
	--sra					Path to SRA accession list
	  or
	--user_reads			        Path to user RNA reads (*_{1,2}.fastq)

Optional arguments:
	--outdir				Path to the output directory (default: easel_results)
	--species				Species prefix (default: SPECIES)
	--chunks				Number of chunks genome will be split into (default: 50)
	--rate					Filtering mapping rate cut-off (default: 85)
	--length				Filtering read length cut-off (default: 15000000)
	--cluster_id			        USEARCH identify threshold (default: 0.80)
	--aligner				Protein alignment tool [gth/miniprot] (default: gth)
	--N					Miniprot: retain at most NUM number of secondary chains/alignments (default: 30)
	--outn					Miniprot: output up to min{NUM, -N} alignments per query (default: 1000)
	--max_intron			        Miniprot: max intron size (default: 200)
	--fastp_cpu				Number of threads dedicated to fastp (default: 4)
	--stringtie_cpu			        Number of threads dedicated to stringtie2 (default: 8)
	--psiclass_cpu			        Number of threads dedicated to psiclass (default: 8)
	--miniprot_cpu			        Number of threads dedicated to miniprot (default: 12)
	--test							Number of genes to test AUGUSTUS
	--train							Number of genes to train AUGUSTUS
	-resume					Resume script from last step 

Documentation:
https://gitlab.com/PlantGenomicsLab/easel-augustus-training

	Authors:
Cynthia Webster <cynthia.webster@uconn.edu>
========================================================================================
```

### Required arguments

`--genome` Path to a repeat masked genome. The reference genome must be masked (hard or soft) to minimize false annotations. We recommend \
`--sra` Path to SRA accession list. The list contains SRR accession ids, for example [SRR17316635](https://www.ncbi.nlm.nih.gov/sra/?term=SRR17316635) is a shotgun whole transcriptome RNAseq experiment. The file should contain one SRR accession id per row in a single column without a header. Transcriptome reads must have minimum quality metrics, including a mapping rate of 85%, 15 million reads (i.e. spots), and a mean read length of 70. The default parameters can be changed in the `nextflow.config` params block.


    rate                     = '85'
    total_reads              = '15000000'
    mean_length              = '70'

`--user_reads` Users with their own sets of RNAseq reads can supply them by providing the path. \


### Optional arguments:
--outdir        Path to the output directory (default: easel_results) \
--species       Species prefix (default: SPECIES) \
--chunks        Number of chunks genome will be split into (default: 50) \
--rate          Filtering mapping rate cut-off (default: 85) \ 
--length        Filtering read length cut-off (default: 15000000) \
--cluster_id    USEARCH identify threshold (default: 0.80) \
--aligner       Protein alignment tool [gth/miniprot] (default: gth) \
--N             Miniprot: retain at most NUM number of secondary chains/alignments (default: 
--outn          Miniprot: output up to min{NUM, -N} alignments per query (default: 1000) \ 
--max_intron    Miniprot: max intron size (default: 200) \
--fastp_cpu     Number of threads dedicated to fastp (default: 4) \ 
--stringtie_cpu Number of threads dedicated to stringtie2 (default: 8) \
--psiclass_cpu  Number of threads dedicated to psiclass (default: 8) \ 
--miniprot_cpu  Number of threads dedicated to miniprot (default: 12) \
--test          Number of genes to test AUGUSTUS \
--train         Number of genes to train AUGUSTUS \
-resume         Resume script from last step  \
